src/openrct2/world/map.c

```diff
@@ -1144,25 +1145,26 @@ static money32 map_clear_scenery_from_tile(sint32 x, sint32 y, sint32 clear, sin
                     return MONEY32_UNDEFINED;
 
                 totalCost += cost;
+
                 if (flags & 1)
                     goto restart_from_beginning;
 
             } break;
         case TILE_ELEMENT_TYPE_WALL:
             if (clear & (1 << 0)) {
-                sint32 eax = x * 32;
-                sint32 ebx = flags;
-                sint32 ecx = y * 32;
-                sint32 edx = (tileElement->base_height << 8) | (tile_element_get_direction(tileElement));
-                sint32 edi = 0, ebp = 0;
-                cost = game_do_command(eax, ebx, ecx, edx, GAME_COMMAND_REMOVE_WALL, edi, ebp);
-
+                cost = wall_remove(x * 32, y * 32, tileElement->base_height, tile_element_get_direction(tileElement), flags);
                 if (cost == MONEY32_UNDEFINED)
                     return MONEY32_UNDEFINED;
 
                 totalCost += cost;
+
+                // NOTE (12/10/2017): This will get us stuck in an infinite loop because game actions are queued
+                // it seems to be trying to remove the same thing over and over.
```

Comment:
When flag is set the tile element would be removed. You have to restart from the beginning to make sure you have deleted every element.

Reply:
I kinda get what its trying but I don't think thats the right way to do it.

Reply:
After a remove tile element the pointers to elements should be recalculated to ensure that the delete has not changed anything being pointed at. As far as I see it it makes sense and there is no better way.

Reply:
This is a rather tricky situation. When you play as a server or client none of the elements are removed right away instead it will happen at the next start of the frame. While playing single player this will removed the elements right away. Would it be okay to ignore the goto if we are playing MP?

Reply:
This is a nested call already. You should be performing the action directly anyway and not sending it over the server. So there is no need to handle it differently.

Reply:
The reason why its queued is to get commands in correct order for everyone. If you remove walls as the server and execute it directly then the other clients might not do this in the same order.  I don't think i can avoid this entirely.

Reply:
I see what you mean by nested but thats not always the case, game_command_clear_scenery uses map_clear_scenery which uses wall_remove.

Now theres game_command_remove_wall which uses wall_remove. 
So it boils down to nested game actions again. 

I was hoping this one has no references from other game commands. So to properly fix this I have to also port game_command_clear_scenery and add the ability for nested actions to get this properly done.

What do you suggest I should do?